File: Algorithms/Implementation/Minimum Distances.php
<?php

/*
 * Complete the 'minimumDistances' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts INTEGER_ARRAY a as parameter.
 */

function minimumDistances($a) {
    # Write your code here
    $d = [];
    foreach ($a as $i => $val_i) {
        foreach ($a as $j => $val_j) {
            if ($i != $j && $val_i == $val_j) {
                $d[] = abs($i - $j);
            }
        }
    }
    if (!empty($d)) {
        return min($d);
    }
    else {
        return -1;
    }
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$n = intval(trim(fgets(STDIN)));
$a_temp = rtrim(fgets(STDIN));
$a = array_map('intval', preg_split('/ /', $a_temp, -1, PREG_SPLIT_NO_EMPTY));
$result = minimumDistances($a);

fwrite($fptr, $result . "\n");
fclose($fptr);

File: Algorithms/Implementation/Minimum Distances.py
import os

#
# Complete the 'minimumDistances' function below.
#
# The function is expected to return an INTEGER.
# The function accepts INTEGER_ARRAY a as parameter.
#

def minimumDistances(a):
    # Write your code here
    d = []
    for i, val_i in enumerate(a):
        for j, val_j in enumerate(a):
            if i != j and val_i == val_j:
                d.append(abs(i - j))
    if d:
        return min(d)
    else:
        return -1

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input().strip())
    a = list(map(int, input().rstrip().split()))

    result = minimumDistances(a)

    fptr.write(str(result) + '\n')
    fptr.close()
